### PR TITLE
[RPA 684] Add internalization to LP

### PR DESCRIPTION
**Pull request by MRSIMM0**
## Description

image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-orchestrator-ui/public/images/flags/en.svg
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/public/images/flags/en.svg:1]: let's remove flags if we dont use them eventually

File: runbotics/runbotics-orchestrator-ui/public/images/flags/world.svg
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/public/images/flags/world.svg:1]: same here, remove it

File: runbotics/package-lock.json
Comment (preludium) [inline runbotics/package-lock.json:1]: something went wrong with installation, remove this file

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/Header/Header.tsx [hunk 1]
```diff
@@ -35,35 +35,49 @@ const Header = () => {
...
+
                 <Navbar
                     isNavExpanded={isNavExpanded}
                     hideNav={hideNav}
                     isMobileVisible={false}
                 />
-                <LoginLink className={styles.loginLink} />
+
+                <div className={styles.buttonGroup}>
+                    <LanguageSwitcher></LanguageSwitcher>
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/Header/Header.tsx]: if not children make it `<LanguageSwitcher />`

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React, { useEffect } from 'react';
+import { useRouter } from 'next/router';
+import styles from './LanguageSwitcher.module.scss';
+import useTranslations from '#src-app/hooks/useTranslations';
+
+import { languages, Language } from '#src-app/translations/translations';
+
+export default function LanguageSwitcher() {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: use TS, I suggest doing it this way `const LanguageSwitcher: VFC = () => {` + default export at the end of the file

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    function capitalizeFirstLetter(string: string) {
+        return string.charAt(0).toUpperCase() + string.slice(1);
+    }
+
+    useEffect(() => {
+        switchLanguage(activeLocale as Language);
+        push(pathname, undefined, {
+            locale: activeLocale,
+        });
+    }, []);
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: add active locale to useEffect deps. Is push required?

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    return (
+        <>
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: no need to have react fragment here

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            <div className={styles.selectWrapper}>
+                <select
+                    onChange={(event) => {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: move event handler to separate function

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+                        switchLanguage(event.target.value as Language);
+                        push(pathname, undefined, {
+                            locale: event.target.value,
+                        });
+                    }}
+                    className={styles.select}
+                >
+                    {languages.map((el) => {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: use full names its not c++ xd. I suggest just `language`

Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: if this arrow function have no body u can make implicite return like this: 
image

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/industries.json
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/industries.json:1]: industries was removed from the project

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/hero.json [hunk 7]
```diff
@@ -1,5 +1,5 @@
 {
-    "Landing.Hero.Title.Part.1": "Keep doing the best",
-    "Landing.Hero.Title.Part.2": "and automate the rest!",
-    "Landing.Hero.Subtitle": "Our mission is to help you to automate your business processes using RPA."
+    "Landing.Hero.Title.Part.1": "Rób to, co najlepsze",
+    "Landing.Hero.Title.Part.2": "i zautomatyzować resztę!",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/hero.json]: these two dont make sense. It should be more like:
1) Rób to, co potrafisz najlepiej
2)  zautomatyzuj resztę!

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/open-source.json [hunk 8]
```diff
@@ -1,13 +1,13 @@
 {
     "Landing.OpenSource.Heading.Title": "Open source",
-    "Landing.OpenSource.Heading.Description.Part.1": "Learn the",
-    "Landing.OpenSource.Heading.Description.Part.2": "undeniable advantages",
-    "Landing.OpenSource.Heading.Description.Part.3": "of open source software",
-    "Landing.OpenSource.Card.1.Title": "Greater security",
+    "Landing.OpenSource.Heading.Description.Part.1": "Dowiedz się",
+    "Landing.OpenSource.Heading.Description.Part.2": "niezaprzeczalne zalety",
+    "Landing.OpenSource.Heading.Description.Part.3": "oprogramowania open source",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/open-source.json]: these 3 are a single sentence. Should be `Poznaj niezaprzeczalne zalety oprogramowania open source`

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/partner.json [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {
-    "Landing.Partner.Title.Part.1": "Our Main Partner",
-    "Landing.Partner.Title.Part.2": "Would you like to become our partner?",
-    "Landing.Partner.Subtitle.Part.1": "We would be happy to welcome YOU on the board!",
-    "Landing.Partner.Subtitle.Part.2": "Contact us to receive more details.",
-    "Landing.Partner.Button.Text": "Become partner"
+    "Landing.Partner.Title.Part.1": "Nasz główny partner",
+    "Landing.Partner.Title.Part.2": "Czy chciałbyś zostać naszym partnerem?",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/partner.json]: skip `Czy`

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/partner.json [hunk 10]
```diff
@@ -1,7 +1,7 @@
...
+    "Landing.Partner.Subtitle.Part.1": "Będziemy z radością powitać CIĘ na pokładzie!",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/partner.json]: Z przyjemnością powitamy Cię w gronie członków!

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/open-source.json [hunk 11]
```diff
@@ -1,13 +1,13 @@
...
+    "Landing.OpenSource.Card.1.Title": "Wyższy poziom bezpieczeństwa",
     "Landing.OpenSource.Card.1.Text": "Lorem Ipsum is simply dummy text of the printing and typesetting industry",
-    "Landing.OpenSource.Card.2.Title": "Free product",
+    "Landing.OpenSource.Card.2.Title": "Bezpłatny produkt",
     "Landing.OpenSource.Card.2.Text": "Lorem Ipsum is simply dummy text of the printing and typesetting industry",
-    "Landing.OpenSource.Card.3.Title.Part.1": "Easy to customize ",
-    "Landing.OpenSource.Card.3.Title.Part.2": "for your own needs",
+    "Landing.OpenSource.Card.3.Title.Part.1": "Łatwy do dostosowania ",
+    "Landing.OpenSource.Card.3.Title.Part.2": "do twoich potrzeb",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/open-source.json]: Łatwe dostosowanie do własnych potrzeb

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/rpa.json [hunk 12]
```diff
@@ -1,10 +1,10 @@
...
-    "Landing.RPA.Content.Paragraph1.Part1": "RPA means Robotics Process Automation. It is the software technology to automate digital tasks, by creating robots that can learn, mimic, and execute rules-based business processes.",
-    "Landing.RPA.Content.Paragraph1.Part2": "Sounds complicated?",
-    "Landing.RPA.Content.Paragraph1.Part3": "It's easy.",
-    "Landing.RPA.Content.Paragraph2": "Just let you're bots see what to do and then let them work for you. RPA bots can interact with any app or system in the same way as you do, but with the difference that they can work all day long, without breaks, faster and more effectively.",
-    "Landing.RPA.Background.ThreeArrows": "Three arrows pointing to the right",
-    "Landing.RPA.Background.OneArrow": "One arrow pointing to the right"
-}
\ No newline at end of file
+    "Landing.RPA.Title.Part1": "Czym jest RPA?",
+    "Landing.RPA.Title.Part2": "I jak go wykorzystujemy, aby dostarczać lepsze rozwiązania?",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/rpa.json]: skip `I`

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss [hunk 13]
```diff
@@ -0,0 +1,53 @@
+@import '#src-landing/scss/main';
+@import '#src-landing/components/Typography/Typography.module.scss';
+
+.option {
+    font-family: 'Montserrat';
+    font-size: large;
+    font-weight: 500;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss]: import it from typography scss

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+    padding: 10px;
+    cursor: pointer;
+}
+.selectWrapper {
+    position: relative;
+    width: 40px;
+    aspect-ratio: 1/1;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss]: why aspect ratio?

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+    display: flex;
+    border: 2px solid black;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss]: dont make border permanent, imo its better to manipulate the background color like search button on blog page. default background is transparent but on hover make it darker than the background. Every color you need is in palette scss

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+    border-radius: 0.25em;
+    font-size: 1.25rem;
+    cursor: pointer;
+    background-color: rgba(255, 255, 255, 0.103);
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss]: colors take from predefined variables from palette scss

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+        background: url('/images/flags/globe.svg') no-repeat;
+        width: 30px;
+        height: 30px;
+        background-size: cover;
+    }
+}
+.select {
+    border-radius: 5px;
+    padding: 10px;
+    z-index: 99;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss]: every z-index is documented in _z-indes.scss. If you need new one then place it there. Maybe you can use existing one +1

**Review by preludium — CHANGES_REQUESTED** [review]
there are many lorem ipsum translations in pl files but these were substituted in en files, adjust translations to match en

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx (hunk 3 above)
Comment (MRSIMM0) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: image

without push only navbar is in polish

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/open-source.json [hunk 18]
```diff
@@ -1,13 +1,13 @@
...
-    "Landing.OpenSource.Card.3.Title.Part.2": "for your own needs",
-    "Landing.OpenSource.Card.3.Text": "Lorem Ipsum is simply dummy text of the printing and typesetting industry"
+    "Landing.OpenSource.Heading.Description.Part.1": "Poznaj",
+    "Landing.OpenSource.Heading.Description.Part.2": "niezaprzeczalne zalety",
+    "Landing.OpenSource.Heading.Description.Part.3": "oprogramowania open source",
+    "Landing.OpenSource.Card.1.Title": "Wyższy poziom bezpieczeństwa",
+    "Landing.OpenSource.Card.1.Text": "Jako produkt open-source, RunBotics oferuje większe bezpieczeństwo dzięki współpracy kierowanej przez społeczność i transparentnemu przeglądowi kodu.",
+    "Landing.OpenSource.Card.2.Title": "Bezpłatny produkt",
+    "Landing.OpenSource.Card.2.Text":"Produkty open-source, takie jak RunBotics, oferują elastyczność, umożliwiając adaptację i modyfikację oprogramowania do konkretnych wymagań, w przeciwieństwie do rozwiązań własnościowych, które oferują ograniczone możliwości dostosowania.",
+    "Landing.OpenSource.Card.3.Title.Part.1": "Łatwe  dostosowanie ",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/open-source.json]: too many spaces

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/hero.json [hunk 19]
```diff
@@ -1,5 +1,5 @@
 {
-    "Landing.Hero.Title.Part.1": "Keep doing the best",
-    "Landing.Hero.Title.Part.2": "and automate the rest!",
-    "Landing.Hero.Subtitle": "Our mission is to help you to automate your business processes using RPA."
+    "Landing.Hero.Title.Part.1": "Rób to, co potrafisz najlepiej",
+    "Landing.Hero.Title.Part.2": "i zautomatyzować resztę!",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/hero.json]: still does not make sense

File: runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/partner.json [hunk 20]
```diff
@@ -1,7 +1,7 @@
 {
-    "Landing.Partner.Title.Part.1": "Our Main Partner",
-    "Landing.Partner.Title.Part.2": "Would you like to become our partner?",
-    "Landing.Partner.Subtitle.Part.1": "We would be happy to welcome YOU on the board!",
-    "Landing.Partner.Subtitle.Part.2": "Contact us to receive more details.",
-    "Landing.Partner.Button.Text": "Become partner"
+    "Landing.Partner.Title.Part.1": "Nasz główny partner",
+    "Landing.Partner.Title.Part.2": "Chciałbyś zostać naszym partnerem?",
+    "Landing.Partner.Subtitle.Part.1": "Łatwe dostosowanie do własnych potrzeb!",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/translations/pl/landing/partner.json:4]: wrong string pasted here

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+
+.selectWrapper {
+
+    margin-top: auto;
+    margin-bottom: auto;
+    width: 35px;
+    height: 35px;
+    position: relative;
+    display: flex;
+    // border: 1px solid $palette-black;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.module.scss]: remove comments + empty spaces

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+    const { switchLanguage, translate } = useTranslations();
+
+    const { push, locale: activeLocale, pathname } = useRouter();
+
+
+
+
+    const capitalizeFirstLetter = (string: string) => string.charAt(0).toUpperCase() + string.slice(1);
+
+    const handleLanguageSwitch = (language: Language) => {
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: too many empty lines

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+                    <option
+                        key={language}
+                      
+                        className={styles.option}
+                        value={language}
+                    >
+                        {capitalizeFirstLetter(
+                            translate(`Common.Languages.${language}`)
+                        )}
+                                &nbsp;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: why nbsp?

**Review by preludium — DISMISSED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx (hunk 23 above)
Comment (MRSIMM0) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: This is walkaround to add padding to option

Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx]: whatever xd

**Pull request closed without merging**